src/main/groovy/wooga/gradle/unity/UnityPlugin.groovy

```diff
@@ -328,4 +330,25 @@ class UnityPlugin implements Plugin<Project> {
             task.group = GROUP
         }
     }
+
+    private static void addAddUPMPackagesTask(Project project, final UnityPluginExtension extension) {
+        def manifestFile = project.layout.projectDirectory.file("Packages/manifest.json")
+        def addUPMPackagesTask = project.tasks.register(Tasks.addUPMPackages.toString(), AddUPMPackages) {task ->
+            task.group = GROUP
+            task.manifestPath.convention(manifestFile)
+            task.upmPackages.putAll(extension.upmPackages)
+            if (extension.enableTestCodeCoverage.getOrElse(false)) {
+                task.upmPackages.put("com.unity.testtools.codecoverage", "1.1.0")
+            }
+        }
+        project.tasks.withType(Test).configureEach {testTask ->
+            testTask.dependsOn(addUPMPackagesTask)
+        }
```

Comment:
```suggestion
        project.tasks.withType(UnityTask).configureEach {task ->
            task.dependsOn(addUPMPackagesTask)
        }
```

I think any task of type Unity would need this if it is requested to have certain upm packages installed or?

Reply:
I don't think so. I believe that for operations like importing the project or generating a solution file this won't be needed. From my point of view, the UPM packages are needed when we do some operation that needs a unity build.